### PR TITLE
Added Event metrics and PDU Session information to sequence diagram

### PR DESCRIPTION
**Pull request by dhruvigoyal25**
This PR makes the following changes to AdvantEDGE:

1. Added Event metrics to sequence metrics.
2. Added PDU session note over information to sequence metrics.

### CODE REVIEW

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 1]
```diff
@@ -589,6 +602,95 @@ func mePostSeqQuery(w http.ResponseWriter, r *http.Request) {
 				continue
 			}
 			t := metricTime.Format("15:04:05.000")
+
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/v2/metrics-engine.go:605]: This function is getting long, please try defining other functions to call from this one.

File: go-apps/meep-metrics-engine/go.mod [hunk 2]
```diff
@@ -12,6 +12,7 @@ require (
 	github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-model v0.0.0
 	github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-mq v0.0.0
 	github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-redis v0.0.0
+	github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-sandbox-ctrl-client v0.0.0-20211214133749-f203f7ab4f1c
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/go.mod:15]: should be added as v0.0.0 with appropriate 'replace' path.

File: go-apps/meep-metrics-engine/go.sum [hunk 3]
```diff
@@ -1,6 +1,8 @@
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
 cloud.google.com/go v0.34.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-sandbox-ctrl-client v0.0.0-20211214133749-f203f7ab4f1c h1:A63lWl3Tgh0XjVmulBpGON5G3hGfy2jpRoZfwl79RN4=
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/go.sum:4]: This should not be in the go.sum file.

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go (hunk 1 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go:605]: Sure, addressed. Added a common function updateSequenceMetrics that updates the response based on the request params: mermaid or sdorg and stronly or listonly.

File: go-apps/meep-metrics-engine/go.mod (hunk 2 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/go.mod]: Addressed.

File: go-apps/meep-metrics-engine/go.sum (hunk 3 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/go.sum]: Updated the file. Addressed.

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 4]
```diff
@@ -556,22 +558,33 @@ func mePostSeqQuery(w http.ResponseWriter, r *http.Request) {
 		duration = params.Scope.Duration
 		limit = int(params.Scope.Limit)
 	}
-	// Get metrics
+	// Get http metrics
+	params.Fields = append(params.Fields, "msg_type", "src", "dst", "endpoint", "body", "method")
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/v2/metrics-engine.go:562]: Use the field constants defined in the metrics package.

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 5]
```diff
@@ -589,28 +602,82 @@ func mePostSeqQuery(w http.ResponseWriter, r *http.Request) {
...
+					}
+					mobilityEventProcessed = true
+
+					// Create group for mobility event
+					eventDescription := eventMetric.Description
+					eventDescription = strings.Replace(eventDescription, "[", "", -1)
+					eventDescription = strings.Replace(eventDescription, "]", "", -1)
+					eventSdorg += "\ngroup Mobility Event: " + eventDescription + "\n"
+					eventMermaid += "note over event: Mobility Event: " + eventDescription + "\n"
+					if params.Fields[0] == "mermaid" {
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/v2/metrics-engine.go:630]: Use the field constants defined in the metrics package.
NOTE: this comment applies to any other hardcoded field strings in this file.

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 6]
```diff
@@ -556,22 +558,33 @@ func mePostSeqQuery(w http.ResponseWriter, r *http.Request) {
...
+	// Get http metrics
+	params.Fields = append(params.Fields, "msg_type", "src", "dst", "endpoint", "body", "method")
 	valuesArray, err := metricStore.GetInfluxMetric(met.HttpLogMetricName, tags, params.Fields, duration, limit)
 	if err != nil {
 		log.Error(err.Error())
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+
 	if len(valuesArray) == 0 {
```
[the remark this answers is not in the document]
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go]: Addressed.

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go (hunk 4 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go]: Updated the code with the field constants for all hardcoded values.

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go (hunk 5 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go]: Addressed.

**Pull request merged**